src/lucky/tags/text_helpers.cr

```diff
@@ -0,0 +1,233 @@
+    truncate(text, length, omission, separator, escape, blk: block)
+  end
+
+  # This could go in a String extension
+  private def truncate_text(text : String, truncate_at : Int32, omission : String = "...", separator : String | Nil = nil)
+    return text unless text.size > truncate_at
+
+    length_with_room_for_omission = truncate_at - omission.size
+    stop = \
+      if separator
+        text.rindex(separator, length_with_room_for_omission) || length_with_room_for_omission
+      else
+        length_with_room_for_omission
+      end
+
+    "#{text[0, stop]}#{omission}"
+  end
+
+  def highlight(text : String, phrases : String | Regex | Array(String | Regex), highlighter : Proc | String = "<mark>\\1</mark>")
+    if text.to_s.blank? || phrases.to_s.blank?
+      text || ""
+    else
+      unless phrases.is_a?(Array)
+        phrase = phrases
+        phrases = Array(String | Regex).new
+        phrases << phrase
+      end
+
+      match = phrases.map do |p|
+        p.is_a?(Regex) ? p.to_s : Regex.escape(p.to_s)
+      end.join("|")
+
+      if highlighter.is_a?(Proc)
+        text.gsub(/(#{match})(?![^<]*?>)/i, &highlighter)
+      else
+        text.to_s.gsub(/(#{match})(?![^<]*?>)/i, highlighter)
+      end
+    end
+  end
+
+  def highlight(text : String, phrases : String | Regex | Array(String | Regex), &block : String -> _)
+    highlight(text, phrases, highlighter: block)
+  end
+
+  def excerpt(text : String, phrase : Regex | String, separator : String = "", radius : Int32 = 100, omission : String = "...")
+    return "" if text.to_s.blank?
+
+    case phrase
+    when Regex
+      regex = phrase
+    else
+      regex = /#{Regex.escape(phrase.to_s)}/i
+    end
+
+    return unless matches = text.match(regex)
+    phrase = matches[0]
+
+    unless separator.empty?
+      text.split(separator).each do |value|
+        if value.match(regex)
+          phrase = value
+          break
+        end
+      end
+    end
+
+    first_part, second_part = text.split(phrase, 2)
+
+    prefix, first_part   = cut_excerpt_part(:first, first_part, separator, radius, omission)
+    postfix, second_part = cut_excerpt_part(:second, second_part, separator, radius, omission)
+
+    affix = [first_part, separator, phrase, separator, second_part].join.strip
+    [prefix, affix, postfix].join
+  end
+
+  def pluralize(count : Int32 | String | Nil, singular : String, plural = nil)
+    word = if (count == 1 || count =~ /^1(\.0+)?$/)
+      singular
+    else
+      plural || LuckyInflector::Inflector.pluralize(singular)
+    end
+
+    "#{count || 0} #{word}"
+  end
+
+  def word_wrap(text : String, line_width : Int32 = 80, break_sequence : String = "\n")
+    text = text.split("\n").map do |line|
+      line.size > line_width ? line.gsub(/(.{1,#{line_width}})(\s+|$)/, "\\1#{break_sequence}").strip : line
+    end
+    text.join(break_sequence)
+  end
+
+  def simple_format(text : String, **html_options, wrapper_tag : String | Symbol = "p")
```

Comment:
I was thinking about this a lot and I think we can remove `content_tag` and the capture helper by passing in an optional block here instead. I think it would be more powerful too:

```crystal
simple_format("my cool test") do |formatted_text|
  p formatted_text, class: "this-is-a-custom-class"
end

# or just `simple_format` if you don't need anything fancy
simple_format("my cool test")
```

Reply:
I like the block syntax a lot. It simplifies the implementation a lot.

I deliberated on this a lot honestly. I didn't like how ugly ```capture``` was then I realized Rails capture implementation is almost verbatim ugly. ```capture``` is useful for a number of things within Rails, but Lucky may be solving for those in other ways.